tests/unit/test_spack.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Test minimal snap."""
+
+import logging
+import pathlib
+import re
+import subprocess
+import unittest
+
+logger = logging.getLogger(__name__)
+
+
+class TestSnap(unittest.TestCase):
+    """Unit test Spack snap."""
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        """Test class setup."""
+        subprocess.run(["tox", "-e", "clean"])
+        logger.info("Building snap")
+        subprocess.run(["tox", "-e", "snap"])
+
+    def test_build(self):
+        """Test snap build status."""
+        snap = subprocess.run(
+            ["tox", "-e", "check"], stdout=subprocess.PIPE, text=True
+        ).stdout.split("\n")
+        ansi_escape = re.compile(r"(\x9B|\x1B\[)[0-?]*[ -\/]*[@-~]")
+        # location differs between local and git runner
+        snap_git = ansi_escape.sub("", snap[1])
+        snap_local = ansi_escape.sub("", snap[2])
+        logger.info("Checking for snap...")
+        self.assertTrue(pathlib.Path(snap_local).exists() or pathlib.Path(snap_git).exists())
+
+    def test_install(self):
+        """Test snap install status."""
+        logger.info("Installing spack snap...")
+        subprocess.run(["tox", "-e", "install"])
+        logger.info("Finished spack snap install!")
+        source = subprocess.run(
```

Comment:
With a test, I don't think we want to swallow errors from the command execution (which this currently does). Perhaps add a check=True in here?

edit: applies throughout this test file, but I'll just make the comment here.